pkg/k8shandler/common.go

```diff
@@ -181,15 +181,30 @@ func newElasticsearchContainer(imageName string, envVars []v1.EnvVar, resourceRe
 	}
 }
 
-func newProxyContainer(imageName, clusterName, namespace string, logConfig LogConfig) (v1.Container, error) {
+func newProxyContainer(spec api.ProxySpec, imageName, clusterName, namespace string, logConfig LogConfig) (v1.Container, error) {
 	cpuLimit, err := resource.ParseQuantity("100m")
 	if err != nil {
 		return v1.Container{}, err
 	}
+	if nil != spec.Resources && !spec.Resources.Limits.Cpu().IsZero() {
+		return v1.Container{}, fmt.Errorf("Not allowed to change es-proxy Cpu limit. It is set to %s", cpuLimit.String())
+	}
 
-	memoryLimit, err := resource.ParseQuantity("64Mi")
-	if err != nil {
-		return v1.Container{}, err
+	var memoryLimit resource.Quantity
+	if nil != spec.Resources && !spec.Resources.Limits.Memory().IsZero() {
```

Comment:
Isnt there a fuction elsewhere to apply defaults when not set?

Reply:
yes, `k8shandler/common.go:newResourceRequirements() : line 399`

Reply:
i couldnt find.

Reply:
that method is for ES container requirements.

Reply:
that is how it is currently used, however there is nothing specific to the elasticsearch container in that function. it is doing a lot of the checking you've added here.

https://github.com/openshift/elasticsearch-operator/blob/master/pkg/k8shandler/common.go#L399-L545